subsys/net/ip/net_context.c

```diff
@@ -1115,6 +1115,27 @@ static int get_context_timepstamp(struct net_context *context,
 #endif
 }
 
+static int get_context_proxy(struct net_context *context,
+			     void *value, size_t *len)
+{
+#if defined(CONFIG_SOCKS)
+	struct sockaddr *addr = (struct sockaddr *)value;
+
+	if (!value || !len) {
+		return -EINVAL;
+	}
+
+	*len = context->options.proxy.addrlen;
+
+	memcpy(addr, &context->options.proxy.addr,
+	       context->options.proxy.addrlen);
```

Comment:
As the *len contains the length of the user specified buffer we want to copy, then the proper sequence would look something like this:
```
	*len = MIN(context->options.proxy.addrlen, *len);

	memcpy(addr, &context->options.proxy.addr, *len);
```